zipline/pipeline/loaders/blaze.py

```diff
@@ -0,0 +1,697 @@
+    single_column = None
+    if isscalar(expr.dshape.measure):
+        # This is a single column, record which column we are to return
+        # but create the entire dataset.
+        single_column = expr._name
+        col = expr
+        for expr in expr._subterms():
+            if isrecord(expr.dshape.measure):
+                break
+        else:
+            expr = bz.Data(col, name=single_column)
+
+    measure = expr.dshape.measure
+    if not isrecord(measure) or AD_FIELD_NAME not in measure.names:
+        raise TypeError(
+            "expr must be a collection of records with at least an '{ad}'"
+            " field. Fields provided: '{fields}'\nhint: maybe you need to use "
+            ' `relabel` to change your field names'.format(
+                ad=AD_FIELD_NAME,
+                fields=measure,
+            ),
+        )
+    _check_datetime_field(AD_FIELD_NAME, measure)
+
+    if TS_FIELD_NAME not in measure.names:
+        expr = bz.transform(expr, **{TS_FIELD_NAME: expr[AD_FIELD_NAME]})
+        if deltas is not None:
+            deltas = bz.transform(
+                deltas,
+                **{TS_FIELD_NAME: deltas[AD_FIELD_NAME]}
+            )
+    else:
+        _check_datetime_field(TS_FIELD_NAME, measure)
+
+    if deltas is not None and (sorted(deltas.dshape.measure.fields) !=
+                               sorted(measure.fields)):
+        raise TypeError(
+            'base measure != deltas measure:\n%s != %s' % (
+                measure,
+                deltas.dshape.measure,
+            ),
+        )
+
+    # Ensure that we have a data resource to execute the query against.
+    _check_resources('expr', expr, resources)
+    _check_resources('deltas', deltas, resources)
+
+    # Create or retrieve the pipeline api dataset.
+    ds = new_dataset(expr, deltas)
+    # Register our new dataset with the loader.
+    (loader if loader is not None else global_loader)[ds] = ExprData(
+        expr,
+        deltas,
+        resources,
+    )
+    if single_column is not None:
+        # We were passed a single column, extract and return it.
+        return getattr(ds, single_column)
+    return ds
+
+
+getdataset = attrgetter('dataset')
+dataset_name = attrgetter('name')
+
+
+def inline_novel_deltas(base, deltas, dates):
+    """Inline any deltas into the base set that would have changed our most
+    recently known value.
+
+    Parameters
+    ----------
+    base : pd.DataFrame
+        The first known values.
+    deltas : pd.DataFrame
+        Overwrites to the base data.
+    dates : pd.DatetimeIndex
+        The dates requested by the loader.
+
+    Returns
+    -------
+    new_base : pd.DataFrame
+        The new base data with novel deltas inserted.
+    """
+    get_indexes = dates.searchsorted
+    return pd.concat(
+        (base,
+         deltas.loc[
+             (get_indexes(deltas[TS_FIELD_NAME].values, 'right') -
+              get_indexes(deltas[AD_FIELD_NAME].values, 'letf')) <= 1
```

Comment:
Pretty sure `'letf'` isn't correct here.  Is this untested, or is pandas doing like "if how.startswith('l')"?